tests/kernel/smp/src/main.c

```diff
@@ -72,20 +89,85 @@ void test_main(void)
 	}
 
 	for (i = 0; i < 10; i++) {
-		/* Wait slightly longer than the other thread so our
-		 * count will always be lower
-		 */
 		k_busy_wait(DELAY_US + (DELAY_US / 8));
-
+		
 		if (t2_count <= i) {
 			ok = 0;
 			break;
 		}
 	}
 
-	if (ok) {
-		TC_END_REPORT(TC_PASS);
-	} else {
-		TC_END_REPORT(TC_FAIL);
+	k_thread_abort(tid);
+
+	zassert_true(ok == 1, "SMP failed");
+}
+
+static void smp_thread_entry(void *p1, void *p2, void *p3)
+{
+	int tnum = (int)p1;
+
+	k_busy_wait(DELAY_US);
+
+	tdata[tnum].executed = 1;
+}
+
+
+/**
+ * @brief Verify round robin scheduling on all cores
+ *
+ * @details The test verifies a scenario in which
+ * parent thread joins children to complete task in
+ * round robin.
+ */
+static void test_round_robin_threads(void)
+{
+	int i, fail = 0;
+
+	/* Make current thread preemptive */
+	k_thread_priority_set(k_current_get(), K_PRIO_PREEMPT(10));
+
+	k_sched_time_slice_set(TIME_SLICE_MS, 0);
+
+	/* Spawn threads of priority equal to
+	 * current thread */
+	for (i = 0; i < THREADS_NUM; i++) {
+		tdata[i].executed = 0;
+		tdata[i].tid = k_thread_create(&tthread[i], tstack[i],
+					       STACK_SIZE, smp_thread_entry,
+					       (void *)i, NULL, NULL,
+					       K_PRIO_PREEMPT(10), 0, 0);
+	}
+
+	k_sleep((DELAY_US * THREADS_NUM) / CONFIG_MP_NUM_CPUS);
```

Comment:
Intention here is to let all threads complete execution in round robin fashion within stipulated time period (time all threads takes to execute / number of cores) . I had given k_busy_wait() instead of k_sleep() with same parameter. In both the cases, the test execution crashes. I don't find flaw in test case. Please help me in this. Also the console output is fragmented, unable to read the output. Possible issue with UART? Need to further debug.